plugins/field-numpad/src/field_numpad.ts

```diff
@@ -0,0 +1,243 @@
+    const key8 = FieldNumpad.createCellInRow_('8', firstRow, 1);
+    const key9 = FieldNumpad.createCellInRow_('9', firstRow, 2);
+    const keyDel = FieldNumpad.createCellInRow_('Del', firstRow, 3);
+
+    const secondRow = table.insertRow(1);
+    const key4 = FieldNumpad.createCellInRow_('4', secondRow, 0);
+    const key5 = FieldNumpad.createCellInRow_('5', secondRow, 1);
+    const key6 = FieldNumpad.createCellInRow_('6', secondRow, 2);
+    const keyClr = FieldNumpad.createCellInRow_('Clr', secondRow, 3);
+
+    const thirdRow = table.insertRow(2);
+    const key1 = FieldNumpad.createCellInRow_('1', thirdRow, 0);
+    const key2 = FieldNumpad.createCellInRow_('2', thirdRow, 1);
+    const key3 = FieldNumpad.createCellInRow_('3', thirdRow, 2);
+
+    const fourthRow = table.insertRow(3);
+    const keySign = FieldNumpad.createCellInRow_('±', fourthRow, 0);
+    const key0 = FieldNumpad.createCellInRow_('0', fourthRow, 1);
+    const keyDecimal = FieldNumpad.createCellInRow_('.', fourthRow, 2);
+
+    key0.addEventListener('click', () => this.onNumClick_('0'));
+    key1.addEventListener('click', () => this.onNumClick_('1'));
+    key2.addEventListener('click', () => this.onNumClick_('2'));
+    key3.addEventListener('click', () => this.onNumClick_('3'));
+    key4.addEventListener('click', () => this.onNumClick_('4'));
+    key5.addEventListener('click', () => this.onNumClick_('5'));
+    key6.addEventListener('click', () => this.onNumClick_('6'));
+    key7.addEventListener('click', () => this.onNumClick_('7'));
+    key8.addEventListener('click', () => this.onNumClick_('8'));
+    key9.addEventListener('click', () => this.onNumClick_('9'));
+    keyDel.addEventListener('click', () => this.onDelClick_());
+    keyClr.addEventListener('click', () => this.onClrClick_());
+    keySign.addEventListener('click', () => this.onSignClick_());
+    keyDecimal.addEventListener('click', () => this.onNumClick_('.'));
+
+    wrapper.appendChild(table);
+    return wrapper;
+  }
+
+  /**
+   * Append the provided numpad digit to the end of the current field value
+   * @param num Numpad digit to append
+   */
+  private onNumClick_(num: string) {
+    const currentValue = this.getText();
+    this.setEditorValue_(currentValue + num);
+  }
+
+  /**
+   * Remove the right most digit from the current field value
+   */
+  private onDelClick_() {
+    const currentValue = this.getText();
+    this.setEditorValue_(currentValue.substring(0,
+        currentValue.length-1));
+  }
+
+  /**
+   * Clear all digits from the current field value
+   */
+  private onClrClick_() {
+    this.setEditorValue_('');
+  }
+
+  /**
+   * Swap the sign of the current field value
+   */
+  private onSignClick_() {
+    const currentValue = this.getText();
+    if (currentValue.startsWith('-')) {
+      this.setEditorValue_(currentValue.substring(1,
+          currentValue.length));
+    } else {
+      this.setEditorValue_(Number('-' + currentValue));
```

Comment:
Why is this editor value being set as a number, but in the positive case above it is being set as a string?  Wouldn't that cause things like `-0` to become `0` and `-1.00` to become `-1`?